integration_tests/src/main/python/cast_test.py

```diff
@@ -304,7 +304,22 @@ def test_cast_array_to_string(data_gen, legacy):
     _assert_cast_to_string_equal(
         data_gen,
         {"spark.sql.legacy.castComplexTypesToString.enabled": legacy})
-
+    
+def test_cast_float_to_string():
```

Comment:
This does not need to be an approximate float?  How many DATAGEN_SEEDS have you tested with this?

Reply:
I tested 10 more times with length=200000 and they all passed. 

The difference between ryu and jdk's float to string is that jdk sometimes keeps more precision, but ryu keeps it as short as possible. The results will be the same float when converted back. So I think it makes sense that we don't need an approximate float here.

Reply:
And also the float to string is fully matched to cpu but double to string is not. It is the reason to test float to string and double to string in two different ways.

Seems the compatibility doc for float to string is outdated, so I'm not sure if we are aware that double to string is not fully matched with cpu. Filed #10037